libs/ledger-live-common/src/config/index.test.ts

```diff
@@ -1,76 +1,29 @@
-import network from "@ledgerhq/live-network/network";
 import { CryptoCurrency } from "@ledgerhq/types-cryptoassets";
 import { getCurrencyConfiguration } from ".";
-import defaultConfig from "./defaultConfig";
-jest.mock("@ledgerhq/live-network/network");
+import liveConfig from "./sharedConfig";
+import { LiveConfig } from "@ledgerhq/live-config/LiveConfig";
 
 describe("getCurrencyConfiguration", () => {
   const cosmosCurrencyMock = {
     family: "cosmos",
     id: "cosmos",
   } as CryptoCurrency;
-
-  const warnSpy = jest.spyOn(console, "warn").mockImplementation(() => {});
-
-  afterEach(() => {
-    getCurrencyConfiguration.reset();
-  });
-
+  LiveConfig.setConfig(liveConfig);
   describe("when config fetching is successful", () => {
     it("should return the related currency config", async () => {
-      const remoteConfigMock = {
-        data: {
-          config: { cosmos: { cosmos: { lcd: "toto" } } },
-        },
-      };
-      // @ts-expect-error method is mocked
-      network.mockResolvedValueOnce(remoteConfigMock);
-      const config = await getCurrencyConfiguration(cosmosCurrencyMock);
-      expect(config).toEqual(remoteConfigMock.data);
-    });
-  });
-
-  describe("when config fetching returns null", () => {
-    beforeEach(() => {
-      // @ts-expect-error method is mocked
-      network.mockResolvedValueOnce("null");
-    });
-
-    it("should return the local currency config and warn the user", async () => {
-      const config = await getCurrencyConfiguration(cosmosCurrencyMock);
-      expect(config).toEqual(defaultConfig.config["cosmos"]["cosmos"]);
-      expect(warnSpy).toHaveBeenCalled();
-    });
-
-    it("should return an error if currency isn't defined in local config", async () => {
-      await expect(
-        getCurrencyConfiguration({
-          id: "idontexistasacurrency",
-          family: "ihavenofamily",
-        } as unknown as CryptoCurrency),
-      ).rejects.toThrow(Error);
+      const config = getCurrencyConfiguration(cosmosCurrencyMock);
+      expect(config).toEqual(liveConfig.cosmos.default);
```

Comment:
I can't mock LiveConfig.getValueByKey output directly because the config lib is not supposed to expose interface directly to change the behaviour of getValueByKey method. 
The only way to change the getValueByKey output is to define a new provider and call the setProvide method to inject this provider since the provider can define its own getValueByKey method. 
This unit test has been written in ledger-live/libs/live-config/src/__tests__/LiveConfig.test.ts (we defined a customised “JsonFileReader” provider in that unit test) so we don't have to test this scenario again here.

In the current test, we don't set the provider and we initialise the config with:
LiveConfig.setConfig(liveConfig);
if the provider is not set, the LiveConfig.getValueByKey just return the default value.
as the getCurrencyConfiguration call the "LiveConfig.getValueByKey", we just test it can retrieve the default value here.

Reply:
Please check this example : https://github.com/LedgerHQ/ledger-live/pull/5829